src/collective/collectionfilter/contentfilter.py

```diff
@@ -9,6 +9,6 @@ def set_content_filter(context, event):
     req = event.request
     if 'collectionfilter' not in req.form:
         return
-    del req.form['collectionfilter']
+    # We leave collectionfilter=1 in req.form so that it gets put into batch links
```

Comment:
The reason for removing the ``collectionfilter`` url parameter was to exit early and not have to ``make_query`` for each traversal step - which can be a lot.
Maybe we should instead check for the existance of ``event.request['contentFilter']`` - if possible?